dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -584,6 +586,17 @@ public AuthenticatedUser createAuthenticatedUser(UserRecordIdentifier userRecord
         return authenticatedUser;
     }
 
+    public AuthenticatedUser createAuthenticatedUser(UserRecordIdentifier userRecordId,
+                                                     String proposedAuthenticatedUserIdentifier,
+                                                     AuthenticatedUserDisplayInfo userDisplayInfo,
+                                                     boolean generateUniqueIdentifier,
+                                                     Locale preferredNotificationsLanguage) {
+        Optional<AuthenticatedUser> authenticatedUser = Optional.ofNullable(createAuthenticatedUser(userRecordId, proposedAuthenticatedUserIdentifier, userDisplayInfo, generateUniqueIdentifier));
+        authenticatedUser.ifPresent(au -> au.setNotificationsLanguage(preferredNotificationsLanguage));
+
+        return authenticatedUser.orElse(null);
```

Comment:
We should never return null, that's what the Optional is for.

Reply:
I can remove optional, but in order to keep the same logic with `createAuthenticatedUser` that is being overloaded, we have to return null if overloaded method returned null.

Reply:
We don't have to keep the same logic, we can refactor some along the way, long term it will be beneficial. If there is a situation where there is complex logic around null, you can always return Option<AuthenticatedUser> and user .getOrNull method in other services. That way your method is clean and you are backwards compatible.